src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileServiceImpl.java

```diff
@@ -64,6 +91,24 @@ public ConcurRequestExtractFileValidationService getConcurRequestExtractFileVali
         return this.concurRequestExtractFileValidationService;
     }
 
+    private String parseRequestExtractFileNameFrom(String requestExtractFullyQualifiedFileName) {
+        //plus 1 because we do not want the forward slash
+        return requestExtractFullyQualifiedFileName.substring((requestExtractFullyQualifiedFileName.lastIndexOf(ConcurConstants.FORWARD_SLASH)) + 1);
+    }
+
+    private ConcurRequestExtractFile loadFileIntoParsedDataObject(String requestExtractFullyQualifiedFileName) {
+        Object parsedFile = loadFile(requestExtractFullyQualifiedFileName);
+        List<ConcurRequestExtractFile> requestExtractFiles = (ArrayList<ConcurRequestExtractFile>) parsedFile;
+        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
+            LOG.error("Single physical file " + requestExtractFullyQualifiedFileName + " should have translated into single parsed file. More or less than one parsed file was detected.");
+            throw new RuntimeException("Single physical file " + requestExtractFullyQualifiedFileName + " should have translated into single parsed file. More or less than one parsed file was detected.");
```

Comment:
do you want to throw runtime exception here? if there are multiple files to be processed and one file fails will these cause the whole batch job to fail?